istio/src/main/scala/io/buoyant/k8s/istio/CurrentIstioPath.scala

```diff
@@ -6,26 +6,65 @@ import com.twitter.logging.Logger
 import io.buoyant.router.context.DstBoundCtx
 
 object CurrentIstioPath {
-  val log = Logger(this.getClass.getName)
-  private val pathLength = 10
+  val logger = Logger(this.getClass.getName)
 
   def apply(maybeBound: Option[Dst.Bound]): Option[Path] = {
     maybeBound.flatMap { bound =>
       bound.id match {
-        case path: Path if (path.elems.length == pathLength) => Some(path)
-        case _ => None
+        case path: Path => Some(path)
+        case other =>
+          logger.warning("Expected path, got %s", other)
+          None
       }
     }
   }
 
+  def targetServiceIn(istioPath: Path): Option[String] = {
+    val pathServiceIndex = istioPath.elems.size - 3
+    if (pathServiceIndex > 0) {
```

Comment:
Should this be `>= 0`?  I mean, element 0 should always be `$` or `#` but I assume this is intended to just avoid index out of bounds.

Reply:
This is pretty much a panic situation so I'm not sure if it matters, but I will add the `=`